lib/gala/payment_token.rb

```diff
@@ -37,6 +39,30 @@ def decrypt(certificate_pem, private_key_pem)
 
     class << self
 
+      def validate_signature(signature, ephemeral_public_key, data, transaction_id, application_data)
+        leaf_cert = nil
+        intermediate_cert = nil
+        p7 = OpenSSL::PKCS7.new(Base64.decode64(signature))
+        p7.certificates.each {|c|
+          c.extensions.each { |e|
+            leaf_cert = c if e.oid == '1.2.840.113635.100.6.29'
+            intermediate_cert = c if e.oid == '1.2.840.113635.100.6.2.14'
```

Comment:
We should probably pull these two oid strings up into a constant like the existing `MERCHANT_ID_FIELD_OID`?